shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -437,6 +441,26 @@ - (void)onPreEngineRestart {
   [self initializeKeyboard];
 }
 
+- (void)notifySemanticsEnabledChanged {
+  BOOL mySemanticsEnabled = !!_bridge;
+  BOOL newSemanticsEnabled = _engine.semanticsEnabled;
+  if (newSemanticsEnabled == mySemanticsEnabled) {
+    return;
+  }
+  if (newSemanticsEnabled) {
+    _bridge = [self createAccessibilityBridgeWithEngine:_engine];
+  } else {
+    // Remove the accessibility children from flutter view before resetting the bridge.
+    _flutterView.accessibilityChildren = nil;
+    _bridge.reset();
+  }
+  NSAssert(newSemanticsEnabled == !!_bridge, @"Failed to update semantics for the view.");
```

Comment:
```suggestion
  NSAssert(newSemanticsEnabled == mySemanticsEnabled, @"Failed to update semantics for the view.");
```

Reply:
But `mySemanticsEnabled` is not automatically updated, and this change is wrong. I can extract `mySemanticsEnabled` to be a getter, but I'm thinking that it's not worth it for all its uses in this single method. What's your opinion?

Reply:
ah nvm, I missed it.